base/src/ast.rs

```diff
@@ -708,94 +708,101 @@ pub fn walk_pattern<'a, V: ?Sized + Visitor<'a>>(v: &mut V, p: &'a Pattern<V::Id
 pub trait Typed {
     type Ident;
 
-    fn env_type_of(&self, env: &TypeEnv) -> ArcType<Self::Ident>;
+    fn env_type_of(&self, env: &TypeEnv) -> ArcType<Self::Ident> {
+        self.try_type_of(env).unwrap()
+    }
```

Comment:
Would be interesting to remove this method, then bubble up the unwraps explicitly - see what is relying on it!

Reply:
I think it may be worth keeping actually. The original assumption with this method is that it is called on something that knows what type it has and in that case it should be infallible (this is the case of parser AST -> core AST translation, only expressions that has been typechecked without error should be passed to translation). However, as the language server now needs to query the type of expression, patterns etc even if the typechecker finds an error it needs a fallible variant of this. I do need to document this assumption though.